### PR TITLE
[PPML] Add schema() method for scala and python  EncryptDataFrameReader

### PR DESCRIPTION
**Pull request by hzjane**
## Description
Add schema() method for scala and python  EncryptDataFrameReader.

### CODE REVIEW

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextPython.scala [hunk 1]
```diff
@@ -70,6 +76,42 @@ class PPMLContextPython[T]() {
...
+  def schema(encryptedDataFrameReader: EncryptedDataFrameReader,
+             value: ClassDict): EncryptedDataFrameReader = {
+    val hashMap: util.HashMap[String, Object] = value.asInstanceOf[util.HashMap[String, Object]]
+    val reslist = new util.LinkedList[StructField]()
+    // get fields list
+    val fieldlist = hashMap.get("fields").asInstanceOf[JList[_]]
+    if (fieldlist == null) {
+      throw new IllegalArgumentException("schema cann't be null")
+    }
+    for (map1 <- fieldlist.asScala.toArray) {
```
Comment (qiyuangong) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextPython.scala:90]: do not use xxx1 etc. change to typeMap

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextPython.scala [hunk 2]
```diff
@@ -70,6 +81,42 @@ class PPMLContextPython[T]() {
              key: String, value: String): EncryptedDataFrameReader = {
     encryptedDataFrameReader.option(key, value)
   }
+  import net.razorvine.pickle.objects.ClassDict
+
+  def schema(encryptedDataFrameReader: EncryptedDataFrameReader,
+             value: ClassDict): EncryptedDataFrameReader = {
+    val hashMap: util.HashMap[String, Object] = value.asInstanceOf[util.HashMap[String, Object]]
```
Comment (qiyuangong) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextPython.scala:88]: changes to fieldsHashMap?

**Review by qiyuangong — APPROVED** [review]
LGTM

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/crypto/dataframe/EncryptedDataFrameReader.scala [hunk 3]
```diff
@@ -38,12 +38,18 @@ class EncryptedDataFrameReader(
...
       keyLoaderManagement: KeyLoaderManagement) {
   protected val extraOptions = new scala.collection.mutable.HashMap[String, String]
+  protected var dataFrameReader = sparkSession.read
 
   def option(key: String, value: String): this.type = {
     this.extraOptions += (key -> value)
     this
   }
 
+  def schema(schema: StructType): this.type = {
```
Comment (qiuxin2012) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/crypto/dataframe/EncryptedDataFrameReader.scala:48]: Add a unit test for this

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/PPMLContext.scala [hunk 4]
```diff
@@ -115,6 +115,10 @@ class PPMLContext {
   def getSparkSession(): SparkSession = {
     sparkSession
   }
+
+  def sql(sqlText: String): DataFrame = {
```
Comment (qiuxin2012) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/PPMLContext.scala:119]: Add unit test for python api.

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/crypto/dataframe/EncryptedDataFrameReader.scala (hunk 3 above)
Comment (hzjane) [reply scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/crypto/dataframe/EncryptedDataFrameReader.scala:48]: Done.

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextPython.scala [hunk 5]
```diff
@@ -70,6 +81,43 @@ class PPMLContextPython[T]() {
...
+             value: ClassDict): EncryptedDataFrameReader = {
+    val classDictMap: util.HashMap[String, Object] =
+      value.asInstanceOf[util.HashMap[String, Object]]
+    val resList = new util.LinkedList[StructField]()
+    // get fields list
+    val fieldList = classDictMap.get("fields").asInstanceOf[JList[_]]
+    if (fieldList == null) {
+      throw new IllegalArgumentException("schema cann't be null")
+    }
+    for ( map <- fieldList.asScala.toArray) {
```
Comment (qiuxin2012) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextPython.scala:96]: val resList = fieldList.asScala.toArray.map{map =>
...
StructField(fieldName, fieldType, nullable = nullable)
}

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextPython.scala [hunk 6]
```diff
@@ -70,6 +81,43 @@ class PPMLContextPython[T]() {
...
+        case "pyspark.sql.types.BooleanType" => BooleanType
+        case "pyspark.sql.types.StringType" => StringType
+        case _ =>
+          throw new IllegalArgumentException(s"Unsupported data type for field $structType")
+      }
+      val structField = StructField(fieldName, fieldType, nullable = nullable)
+      resList.add(structField)
+    }
+    val schema = StructType(resList)
+    println("Schema: " + schema)
```
Comment (qiuxin2012) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/python/PPMLContextPython.scala:118]: delete this print

File: scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/PPMLContext.scala (hunk 4 above)
Comment (Uxito-Ada) [inline scala/ppml/src/main/scala/com/intel/analytics/bigdl/ppml/PPMLContext.scala:119]: does this sql method support crypto IO?

**Pull request merged**